packages/nuxt/src/app/composables/ssr.ts

```diff
@@ -56,3 +57,10 @@ export function prerenderRoutes (path: string | string[]) {
   const paths = toArray(path)
   appendHeader(useRequestEvent(), 'x-nitro-prerender', paths.map(p => encodeURIComponent(p)).join(', '))
 }
+
+export const setInterval = import.meta.client ? window.setInterval : () => {
+  throw createError({
```

Comment:
In production, perhaps don't throw an error, but just do nothing?

```suggestion
  if (!import.meta.dev) {
    return
  }
  throw createError({
```

Reply:
I would suggest to show a warning in production as well. If anyhow setInterval slips away it can cause memory leaks that are more critical in production.

Reply:
That way at least any kind of error tracking (or QA) could catch it 👍

Reply:
Well if we disable only in dev, we could still hang `nuxt generate`.

It might be tough, but they should never use it on ssr anyway

Reply:
Good idea about generate. For prerender only (`import.meta.prerender`) we can directly return

Reply:
what would be the use case of allowing when generating?

Reply:
Only if suggestion in this comment (dev ~> return) i said (dev||prerender ~> return). Since it is an auto import only for Nuxt apps, i also agree with always throwing 👍🏼

Reply:
ah, my suggestion was to throw error _only_ in dev